lib/fuse.c

```diff
@@ -1272,11 +1315,20 @@ static int try_get_path2(struct fuse *f, fuse_ino_t nodeid1, const char *name1,
 static int get_path2(struct fuse *f, fuse_ino_t nodeid1, const char *name1,
 		     fuse_ino_t nodeid2, const char *name2,
 		     char **path1, char **path2,
-		     struct node **wnode1, struct node **wnode2)
+		     struct node **wnode1, struct node **wnode2,
+		     bool dir_loop)
 {
 	int err;
 
 	pthread_mutex_lock(&f->lock);
+
+	if (dir_loop)
```

Comment:
This looks like more churn than necessary. Is there a reason not to omit the new parameter and replace the `if(dir_loop)` with `ifdef CHECK_DIR_LOOP`?

Reply:
I agree. I struggled with the question of how to properly handle the conditional and decided to submit to get your feedback.

Are you ok with keeping the `dir_loop` parameter passed to `get_path2`, but then being ignored on Linux? Like so:

```C
#if defined(CHECK_DIR_LOOP)
static int check_dir_loop(struct fuse *f,
			  fuse_ino_t nodeid1, const char *name1,
			  fuse_ino_t nodeid2, const char *name2)
{
			  ...
}
#endif

...

#if defined(CHECK_DIR_LOOP)
	if (dir_loop)
	{
		err = check_dir_loop(f, nodeid1, name1, nodeid2, name2);
		if (err)
			goto out_unlock;
	}
#else
	(void)dir_loop;
#endif
```